File: 1089-Duplicate-Zeros.java
class Solution {
    public void duplicateZeros(int[] arr) 
    {
        int size = arr.length;
        for (int i = 0; i < size - 1; i++) 
        {
            if (arr[i] == 0) 
            {
                for (int j = size - 1; j > i; j--) 
                {
                    arr[j] = arr[j - 1];
                }
                arr[i + 1] = 0;
                i++;
            }
        }
    }
}

File: 1089-Duplicate-Zeros.cpp

class Solution {
    public:

    void duplicateZeros(vector<int>& arr)
    {

        int size = arr.size();
        vector<int> vec ;
        for (int i = 0; i < size; i++) 
        {
            if (arr[i] == 0) 
            {
                vec.push_back(0) ;
                vec.push_back(0) ;
            }
            else
            {
                vec.push_back(arr[i]) ;
            }
        }
        for(int i = 0 ; i < size ; i ++)
        {
            arr[i] = vec[i] ;
        }
    }
};